src/System.Composition.Convention/src/Microsoft/Internal/Lock.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.Internal
+{
+    internal sealed class Lock : IDisposable
+    {
+#if FEATURE_SLIMLOCK
```

Comment:
ReaderWriterLockSlim does exist in CoreCLR, so will FEATURE_SLIMLOCK always be defined, and if so, can we get rid of the rest of the dead code?